new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java

```diff
@@ -816,96 +1147,177 @@ private static String createAndVerifyDomainImage() {
+
+    return wdtImage;
+  }
+
+  //create domain from provided image and verify it's start
+  private static void createAndVerifyDomain(String miiImage,
+                                            String domainUid,
+                                            String namespace,
+                                            String domainHomeSource,
+                                            int replicaCount) {
     // create secret for admin credentials
     logger.info("Create secret for admin credentials");
     String adminSecretName = "weblogic-credentials";
-    assertDoesNotThrow(() -> createSecretWithUsernamePassword(adminSecretName, domain1Namespace,
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(adminSecretName, namespace,
         "weblogic", "welcome1"),
         String.format("create secret for admin credentials failed for %s", adminSecretName));
 
     // create encryption secret
     logger.info("Create encryption secret");
     String encryptionSecretName = "encryptionsecret";
-    assertDoesNotThrow(() -> createSecretWithUsernamePassword(encryptionSecretName, domain1Namespace,
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(encryptionSecretName, namespace,
         "weblogicenc", "weblogicenc"),
         String.format("create encryption secret failed for %s", encryptionSecretName));
 
     // create domain and verify
     logger.info("Create model in image domain {0} in namespace {1} using docker image {2}",
-        domainUid, domain1Namespace, miiImage);
-    createDomainCrAndVerify(adminSecretName, REPO_SECRET_NAME, encryptionSecretName, miiImage,domainUid);
-
+        domainUid, namespace, miiImage);
+    createDomainCrAndVerify(adminSecretName, REPO_SECRET_NAME, encryptionSecretName, miiImage,domainUid,
+            namespace, domainHomeSource, replicaCount);
+    String adminServerPodName = domainUid + "-admin-server";
     // check that admin server pod exists in the domain namespace
     logger.info("Checking that admin server pod {0} exists in namespace {1}",
-        adminServerPodName, domain1Namespace);
-    checkPodExists(adminServerPodName, domainUid, domain1Namespace);
+        adminServerPodName, namespace);
+    checkPodExists(adminServerPodName, domainUid, namespace);
 
     // check that admin server pod is ready
     logger.info("Checking that admin server pod {0} is ready in namespace {1}",
-        adminServerPodName, domain1Namespace);
-    checkPodReady(adminServerPodName, domainUid, domain1Namespace);
+        adminServerPodName, namespace);
+    checkPodReady(adminServerPodName, domainUid, namespace);
 
     // check that admin service exists in the domain namespace
     logger.info("Checking that admin service {0} exists in namespace {1}",
-        adminServerPodName, domain1Namespace);
-    checkServiceExists(adminServerPodName, domain1Namespace);
+        adminServerPodName, namespace);
+    checkServiceExists(adminServerPodName, namespace);
```

Comment:
this has to be done before podReady, pod will never become ready when service creation fails.